zanata-war/src/main/java/org/zanata/action/JavaScriptTestHelper.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2014, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.action;
+
+import lombok.extern.slf4j.Slf4j;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Named;
+
+/**
+ * @author Sean Flanigan <a href="mailto:sflaniga@redhat.com">sflaniga@redhat.com</a>
+ */
+
+@Named("javaScriptTestHelper")
+@Dependent
+@Slf4j
+public class JavaScriptTestHelper {
```

Comment:
On 1 March 2016 at 09:43, Carlos Munoz <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/action/JavaScriptTestHelper.java
> <https://github.com/zanata/zanata-server/pull/1072#discussion_r54499942>:
>
> > + */
> > +package org.zanata.action;
> > +
> > +import lombok.extern.slf4j.Slf4j;
> > +
> > +import javax.enterprise.context.Dependent;
> > +import javax.inject.Named;
> > +
> > +/**
> > + * @author Sean Flanigan <a href="mailto:sflaniga@redhat.com">sflaniga@redhat.com</a>
> > + */
> > +
> > +@Named("javaScriptTestHelper")
> > +@Dependent
> > +@Slf4j
> > +public class JavaScriptTestHelper {
>
> should this class go in the test directory instead?
>
No, because it needs to be built into the war for use in functional test
deployments.


> should it be a deactivated alternative?
>

Perhaps so, but it's really just a better named AjaxCounterBean, and at
this point I'd rather leave it alone than fight with alternatives and
activation in zanata-test-war.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1072/files#r54499942>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat